### ISSUE TITLE
unable to run PlexBackup

### ISSUE THREAD

**Issue by alamoudimoh** [alekdavis/PlexBackup#35]
Hi, i am not good with powershell, so what i did it as follows:

1. downloaded the files
2. Right clicked on PlexBackup, Run with PowerShell
3. got the following message
Execution Policy Change
The execution policy helps protect you from scripts that you do not trust. Changing the execution policy might expose
you to the security risks described in the about_Execution_Policies help topic at
https:/go.microsoft.com/fwlink/?LinkID=135170. Do you want to change the execution policy?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): A

i selected Y
the powershell immediately closes and even if i re-open it again, it keeps closing

also, how do i setup the location, do i need to do anything? or it will automatically make the folders in the same location?

EDIT: i think i managed to run it, i was doing it all wrong, now i got the following error, which i think cuz i did not set up the backup directory, which i really do it what to do with

PS C:\Users\MAIN\OneDrive\AppsBackups\Plex> .\PlexBackup.ps1
PlexBackup v1.7.2 (c) 2020 Alek Davis
Script started at:
  07/25/2020 12:22:05
Cannot initialize run-time configuration settings.
PS C:\Users\MAIN\OneDrive\AppsBackups\Plex>

i would like all the backups to be stored in C:\Users\MAIN\OneDrive\AppsBackups\Plex (it is my onedrive folder)

**Comment by alekdavis** [comment]
Did you follow the instructions on the [Getting Started](https://github.com/alekdavis/PlexBackup/blob/master/GETTING%20STARTED.md) page?

**Comment by alamoudimoh** [comment]
> Did you follow the instructions on the [Getting Started](https://github.com/alekdavis/PlexBackup/blob/master/GETTING%20STARTED.md) page?

Yes, 

1. Saved files in the same location
2. Rename the file name and removed SAMPLE
3. Created a powershell shortcut on desktop and in advanced i checked run as admin
4. CD the file directory
5. Executed .\PlexBackup.ps1 and i got the error shared above


I do not what to do with update config File setting. It is not clear to me

**Comment by alekdavis** [comment]
Try adding this line inside of the `catch` block on line 3216:

```
LogException $_ $false $false
```

So the code block will look like this:
```
# Load config settings from a config file (if any).
try {
    LogException $_ $false $false
    Import-ConfigFile -ConfigFilePath $ConfigFile -DefaultParameters $PSBoundParameters
}
catch {
    LogWarning "Cannot initialize run-time configuration settings." $false
    $mutex = StopSingleInstance $mutex
    exit $EXITCODE_ERROR_CONFIG
}
```

**Comment by alekdavis** [comment]
Sorry, messed up the code block (updated now). Basically, I want to see the root cause error message. I suspect, there is something wrong with the config file or (less likely) the module that is supposed to process the config file. Please post the error message after you run the script.

**Comment by deanalanmills** [comment]
alekdavis, above  you say "updated now" but looks like same file with date time stamp the same in the zip file. What am I missing?

thanks for all you have done

**Comment by alekdavis** [comment]
I updated my comment (originally added the new line in the wrong place), not the script itself.

**Comment by alamoudimoh** [comment]
> Sorry, messed up the code block (updated now). Basically, I want to see the root cause error message. I suspect, there is something wrong with the config file or (less likely) the module that is supposed to process the config file. Please post the error message after you run the script.


apologies for the delay i was having an emergency, here you go..


PS C:\Users\MAIN\OneDrive\AppsBackups\Plex> .\PlexBackup.ps1
PlexBackup v1.7.2 (c) 2020 Alek Davis
Script started at:
  07/29/2020 22:19:13
LogException : Invalid JSON primitive: .
At C:\Users\MAIN\OneDrive\AppsBackups\Plex\PlexBackup.ps1:3216 char:5
+     LogException $_ $false $false
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,LogException

Cannot initialize run-time configuration settings.

**Comment by alekdavis** [comment]
Yeah, looks like your configuration file is messed up. Can you post it here?

**Comment by alamoudimoh** [comment]
> Yeah, looks like your configuration file is messed up. Can you post it here?

I just saved the files from https://github.com/alekdavis/PlexBackup/blob/master/PlexBackup.ps1

here is the file
[PlexBackup.zip](https://github.com/alekdavis/PlexBackup/files/4997237/PlexBackup.zip)

**Comment by alekdavis** [comment]
Not this file. I am asking for PlexBackup.ps1.json

**Comment by alamoudimoh** [comment]
> Not this file. I am asking for PlexBackup.ps1.json

[PlexBackup.ps1.zip](https://github.com/alekdavis/PlexBackup/files/4997570/PlexBackup.ps1.zip)

**Comment by alekdavis** [comment]
It looks like you save the web page layout instead of the configuration file. Just download the PlexBackup.ps1.SAMPLE.json (and make sure you download the actual file, not the web page), rename it (remove the .SAMPLE part, put the proper value in the backup directory setting (make sure you use double back slashes for each back slash in the path, so in your case, the value would be `C:\\Users\\MAIN\\OneDrive\\AppsBackups\\Plex` and try it again. Basically, follow the instructions in the Getting Started section.

**Comment by alamoudimoh** [comment]
> It looks like you save the web page layout instead of the configuration file. Just download the `PlexBackup.ps1.SAMPLE.json` (and make sure you download the actual file, not the web page), rename it (remove the `.SAMPLE` part), put the proper value in the backup directory setting (make sure you use double back slashes for each back slash in the path, so in your case, the value would be `C:\\Users\\MAIN\\OneDrive\\AppsBackups\\Plex`) and try it again. Basically, follow the instructions in the Getting Started section.

thanks for the help, i was doing everything wrong :(

ok, i managed to correct it and i can see the progress is running fine except there is an error with registry

  C:\Users\MAIN\OneDrive\AppsBackups\Plex\20200731035027\3\Plex2.reg
LogException : ERROR: The system was unable to find the specified registry key or value.
At C:\Users\MAIN\OneDrive\AppsBackups\Plex\PlexBackup.ps1:3118 char:13
+             LogException $_
+             ~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,LogException

Restoring special subfolders.
Copying folder:
  C:\Users\MAIN\OneDrive\AppsBackups\Plex\20200731035027\4\Plug-in Support\Data\com.plexapp.system\DataItems\Deactivated
to:
  C:\Users\MAIN\AppData\Local\Plex Media Server\Plug-in Support\Data\com.plexapp.system\DataItems\Deactivated
at:
  2020/07/31 04:11:29.830
Completed at:
  2020/07/31 04:11:29.850
Starting Plex service(s):
  Plex Update Service
Starting Plex Media Server process:
  C:\Program Files (x86)\Plex\Plex Media Server\Plex Media Server.exe
Script ended at:
  2020/07/31 04:11:30.324
Script ran for (hr:min:sec.msec):
  00:21:03.192
Script returned:
  SUCCESS
Done.
PS C:\Users\MAIN\OneDrive\AppsBackups\Plex>

[Backup.err.log](https://github.com/alekdavis/PlexBackup/files/5004504/Backup.err.log)

**Comment by alekdavis** [comment]
Yes, you can ignore it for now. Just make sure your backup files are of reasonable size. I will handle the missing registry key better later. Also check the Plex1.reg file to make sure it's backed up.

**Comment by alekdavis** [comment]
Fixed the registry issue in v2.0.0.
